apps/filebrowser/src/filebrowser/tasks.py

```diff
@@ -121,63 +136,105 @@ def document_cleanup_task(**kwargs):
-          elif os.path.isdir(file_path):
-            shutil.rmtree(file_path)
-          LOG.info(f"Deleted {file_path}")
-      except PermissionError:
-        LOG.warning(f"Permission denied: unable to delete {file_path}")
-      except Exception as err:
-        check_disk_usage_and_clean_task.update_state(task_id=task_id, state='FAILURE', meta=kwargs)
-        raise Exception(f"Failed to delete {file_path}. Reason: {err}")
-
+    delete_old_files(tmp_dir, now, timedelta(minutes=60))
     kwargs["progress"] = "100%"
-    check_disk_usage_and_clean_task.update_state(task_id=task_id, state='SUCCESS', meta=kwargs)
+    kwargs["task_end"] = timezone.now().astimezone(pytz.timezone(TIME_ZONE)).strftime("%Y-%m-%dT%H:%M:%S.%f")
+    check_disk_usage_and_clean_task.update_state(task_id=task_id, state=states.SUCCESS, meta=kwargs)
     LOG.info("/tmp directory cleaned.")
-
   else:
     kwargs["progress"] = "100%"
-    check_disk_usage_and_clean_task.update_state(task_id=task_id, state='SUCCESS', meta=kwargs)
+    kwargs["task_end"] = timezone.now().astimezone(pytz.timezone(TIME_ZONE)).strftime("%Y-%m-%dT%H:%M:%S.%f")
+    check_disk_usage_and_clean_task.update_state(task_id=task_id, state=states.SUCCESS, meta=kwargs)
     LOG.info(f"Disk usage is {disk_usage.percent}%, no need to clean up.")
 
   # Get available disk space after cleanup
   free_space = psutil.disk_usage('/tmp').free
   return {'free_space': free_space}
+
+
+@app.task()
+def cleanup_stale_uploads(**kwargs):
+  timeout_minutes = kwargs.get("timeout_minutes", TASK_SERVER_V2.CLEANUP_STALE_UPLOADS_IN_REDIS_PERIODIC_INTERVAL.get())
+  username = kwargs.get("username", "celery_scheduler")
+  task_id = kwargs.get("qquuid")
+  kwargs = {
+    "username": username,
+    "task_name": "cleanup_stale_uploads",
+    "task_id": task_id,
+    "parameters": timeout_minutes,
+    "progress": "0%",
+    "task_start": timezone.now().astimezone(pytz.timezone(TIME_ZONE)).strftime("%Y-%m-%dT%H:%M:%S"),
+  }
+  cleanup_stale_uploads.update_state(task_id=task_id, state=states.STARTED, meta=kwargs)
+  redis_client = parse_broker_url(TASK_SERVER_V2.BROKER_URL.get())
+  try:
+    current_time = int(datetime.now().timestamp())
+
+    for key in redis_client.scan_iter('upload__*'):
+      timestamp_key = f'{key.decode()}_timestamp'
+      timestamp = redis_client.get(timestamp_key)
+
+      if timestamp:
+        timestamp = int(timestamp)
+        if current_time - timestamp > timeout_minutes * 60:
+          file_size = int(redis_client.get(key))
+          redis_client.incrby('upload_available_space', file_size)
+          redis_client.delete(key)
+          redis_client.delete(timestamp_key)
+    initialize_free_disk_space_in_redis()
+    kwargs["progress"] = "100%"
+    kwargs["task_end"] = timezone.now().astimezone(pytz.timezone(TIME_ZONE)).strftime("%Y-%m-%dT%H:%M:%S.%f")
+    cleanup_stale_uploads.update_state(task_id=task_id, state=states.SUCCESS, meta=kwargs)
+  finally:
```

Comment:
where is "except" condition logic? if you are using "try" and "finally" then one assumes it may break and catch in "except" logic.

Reply:
Thanks for catching it. Added "except" logic.